### ISSUE TITLE
Questions about RIGBD implementation and evaluation details in SPEAR

### ISSUE THREAD

**Issue by Matilda0224** [yhDing/SPEAR#6]
Dear authors,
Thank you for releasing the SPEAR code. I have several technical questions regarding the RIGBD implementation and evaluation:

1.  In your implementation of RIGBD (see GCN.py, lines 101–103), the code is:
if finetune == True:
      self.fientune(self.labels, idx_train, idx_val, attach, train_iters, verbose)
self._train_with_val(self.labels, idx_train, idx_val, train_iters, verbose)
**However, the correct RIGBD implementation should be:**
if finetune == True:
      self.fientune(self.labels, idx_train, idx_val, attach, train_iters, verbose)
else:
      self._train_with_val(self.labels, idx_train, idx_val, train_iters, verbose)
Does this modification remove the robust training phase, thereby potentially leading to higher ASR in your reported results?

2.  For Clean Accuracy (CA) calculation, most prior works (e.g., UGBA, DPGBA, and RIGBD) use induct_x as input rather than poison_x. In your implementation, CA is computed on poison_x.
In your paper, CA is described as “clean accuracy (CA), which measures the classification accuracy of the backdoored GNN on clean nodes.”
Could you please clarify whether this is consistent with your code?

3.  On the PubMed dataset, I noticed that in main_rigbd, setting defense_mode=prune is required to reproduce the paper-reported results. Without enabling pruning, RIGBD’s defense performance appears much better.
Could you explain why this is the case? Is there any relevant explanation in the paper?

Thanks a lot for clarifying!

**Comment by yhDing** [comment]
Hi, thanks for the careful check and thoughtful questions!

For the finetune branch, you’re right and we’ll clean that up to match the intended logic.

For CA, the clean accuracy (CA) evaluates how the model trained on poisoned samples performs on clean test data. Since poison_x represents the feature with poisoned training samples, it is the appropriate choice for computing CA (you may refer to #4 ). We notice that the released script accidentally kept defense_mode=prune on, please keep follow the setting in our paper. Your finding is quite worth thinking about: a defense-style pruning step can sometimes make the backdoor more unnoticeable. 

We will push a small fix to keep the released code consistent with the paper settings.

Thanks again — these comments really help improve reproducibility.

**Comment by Matilda0224** [comment]
Thank you for the clarification!

1. CA: You’re absolutely right — I double-checked and realized that UGBA and DPGBA also compute CA with poison_x. My mistake there, thanks for pointing it out.
2. After adding the else branch in the finetune method, I wasn’t able to fully reproduce the results reported in the paper. If possible, could you please share the results you obtained after this fix (or the exact configuration used)? That would really help ensure consistency in replication.
For reference, here are my results of _main_rigbd.py_:

- Cora: before defense (ASR: 0.9815, CA: 0.8370) → after defense (ASR: 0.0517, CA: 0.8296)
- Pubmed: before defense (ASR: 0.9670, CA: 0.8564) → after defense (ASR: 0.9752, CA: 0.8493)
- OGB-Arxiv: before defense (ASR: 0.9648, CA: 0.6694) → after defense (ASR: 0.9548, CA: 0.6604)

It seems that except for Cora, the attack can indeed bypass the RIGBD defense, while on Cora the defense is quite effective. I currently have no clear explanation for this discrepancy — could you possibly share any insights or hypotheses about why this might happen? I’d greatly appreciate your thoughts.

Thanks again for your detailed responses!

**Comment by yhDing** [comment]
Thanks for sharing the results. Cora is indeed a rather special dataset — it’s small in scale and highly homophilous. In fact, I also spent the most time tuning on Cora — perhaps trying a broader range of hyperparameters could help.
